invenio_oaiserver/config.py

```diff
@@ -133,3 +133,6 @@
 
 OAISERVER_CELERY_TASK_CHUNK_SIZE = 100
 """Specify the maximum number of records each task will update."""
+
+OAISERVER_CONTROL_NUMBER_KEY = 'control_number'
```

Comment:
I would keep the model assumptions simple, i.e. the control number has to be in metadata prior to minting (recid or in case of Zenodo a DOI minter assume only that), but I perhaps a more generic "key fetcher" could be useful here instead of assuming that control number resides at level 0 of the dict.